src/scenes/UserProfile/UserProfile.jsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Link } from 'react-router-dom'
+
+import NotFound from 'scenes/NotFound'
+import Loading from 'components/Loading'
+import Card, { CardText } from 'components/Card'
+
+import { getAllPersonalData } from './queries'
+
+const UserProfile = props => {
+  if (!props.data) {
+    props.history.push('/auth/login')
+  }
+
+  const { loading, error, User } = props.data
+
+  if (loading) {
+    return <Loading />
+  }
+
+  if (error) {
+    return <div>Error :(</div>
+  }
+
+  if (User === null) {
+    return <NotFound {...props} />
```

Comment:
If `User === null`, we should show an error instead of a 'NotFound', as this isn't a normal case